### ISSUE TITLE
JS/C++ validator discrepancy

### ISSUE THREAD

**Issue by cramforce** [ampproject/amphtml#2040]
Having `<amp-pixel>` inside of `<head>` fails validation on the cache, but passes JS validation. The cache is correct.

Example URL
http://output.jsbin.com/ruguqu/quiet
https://cdn.ampproject.org/c/output.jsbin.com/ruguqu/quiet?2

**Comment by rudygalfi** [comment]
This is overdue on the Feb 19 milestone. Any update?

**Comment by Gregable** [comment]
tl;dr version: This isn't fixed fully, but the primary symptom is no longer present. Documents that fail cache validation in this way also now fail j/s validation. 

tl version:
The remaining issue is that they fail with different error messages. The j/s validation error will be something about `<amp-pixel>` not allowed inside `<head>`. The cache validation error will usually complain about some tag found after the `<amp-pixel>` being allowed only inside `<head>`. The cache validator is closing the head before `<amp-pixel>`, whereas the javascript validator is just not allowing  `<amp-pixel>` in the `<head>`.

My recommendation would be to remove this from the milestone, but leave the issue open. No strong opinion as we're also tracking this at b/27128311

**Comment by rudygalfi** [comment]
Thanks for clarifying @Gregable!

**Comment by Gregable** [comment]
This is fully fixed now. Closing.
